nematode.js

```diff
@@ -0,0 +1,95 @@
+
+/**
+ * Bibite
+ * 
+ * A bibite is a creature that can see food and move towards it.
+ * It has three eyes that can see food. The eyes are raycasted from the bibite's position.
+ * The bibite has a neural network that takes the eye raycasts as inputs and outputs a rotation and speed.
+ * The bibite moves in the direction of the rotation and at the speed.
+ */
+
+// The maximum distance that the bibite can see
+var maxSeeDistance = 500;
+
+// DEBUG: Should the eye rays be drawn?
+var drawEyeRays = false;
+
+class Nematode {
+
+    constructor() {
+
+        this.nn = null;                         // new NeuralNetwork(3, 2);  // The neural network of the Nematode
+
+        this.age = 0;                           // The age in seconds
+        this.energy = 100;                      // The energy of the Nematode
+
+        this.maxSpeed = 3;                     // The maximum speed of the Nematode
+        this.maxTurnSpeed = 4;                  // The maximum turn speed of the Nematode
+
+        this.width  = 40;                       // The width of the Nematode
+        this.height = 40;                       // The height of the Nematode
+        this.angle = Math.random() * 360;       // The angle of the Nematode
```

Comment:
if you think it's easier: possibly encode the direction as a vector2, the magnitude of which is the current speed of the nematode.